src/test/java/com/commercetools/sync/products/helpers/ProductReferenceResolverTest.java

```diff
@@ -70,72 +74,84 @@ public void resolveProductTypeReference_WithKeysAsUuidSetAndAllowed_ShouldResolv
         final ProductSyncOptions productSyncOptions = ProductSyncOptionsBuilder.of(mock(SphereClient.class))
                                                                                .setAllowUuidKeys(true)
                                                                                .build();
-        final ProductDraft productDraft = mock(ProductDraft.class);
-        when(productDraft.getProductType()).thenReturn(ProductType.referenceOfId(UUID.randomUUID().toString()));
-
+        final ProductDraftBuilder productBuilder = getProductDraftWithRandomRefId();
 
         final ProductReferenceResolver productReferenceResolver = new ProductReferenceResolver(productSyncOptions,
             productTypeService, categoryService, typeService, channelService, taxCategoryService, stateService);
 
-        final ProductDraft resolvedDraft = productReferenceResolver.resolveProductTypeReference(productDraft)
+        final ProductDraftBuilder resolvedDraft = productReferenceResolver.resolveProductTypeReference(productBuilder)
                                                                    .toCompletableFuture().join();
 
-
         assertThat(resolvedDraft.getProductType()).isNotNull();
         assertThat(resolvedDraft.getProductType().getId()).isEqualTo(PRODUCT_TYPE_ID);
     }
 
+    @Nonnull
+    private static ProductDraftBuilder getProductDraftWithRefId(@Nonnull final String refId) {
+        return ProductDraftBuilder.of(ProductType.referenceOfId(refId),
+            LocalizedString.ofEnglish("testName"),
+            LocalizedString.ofEnglish("testSlug"),
+            (ProductVariantDraft)null);
+    }
+
+    @Nonnull
+    private static ProductDraftBuilder getProductDraftWithRef(@Nonnull final Reference<ProductType> reference) {
+        return ProductDraftBuilder.of(reference,
+            LocalizedString.ofEnglish("testName"),
+            LocalizedString.ofEnglish("testSlug"),
+            (ProductVariantDraft)null);
+    }
+
+    @Nonnull
+    private static ProductDraftBuilder getProductDraftWithRandomRefId() {
```

Comment:
@andrii-kovalenko-ct I would name it: `getProductDraftWithRandomProductTypeUUID` it needs to be apparent that it sets a random uuid for the product type

Reply:
3552d9a6